cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginService.java

```diff
@@ -129,11 +193,9 @@ private PluginEndpoint getPluginEndpoint(NamespaceId namespace, ArtifactDetail a
     }
 
     // initialize parent classloader and plugin instantiator
-    parentClassLoader = artifactRepository.createArtifactClassLoader(parentArtifactDescriptor.getLocation());
-    pluginInstantiator = new PluginInstantiator(cConf, parentClassLoader, stageDir);
-
-    // add plugin artifact
-    pluginInstantiator.addArtifact(artifactDetail.getDescriptor().getLocation(), artifactId.toArtifactId());
+    Instantiators instantiators = this.instantiators.getUnchecked(parentArtifactDescriptor);
```

Comment:
if we do it this way, updates to snapshot artifacts will not be seen.  To do this properly we probably have to start storing a timestamp for when an artifact was added, and create a new instantiator if the timestamp is newer than the one we have cached.